### PR TITLE
Bail out of the loop when the stub Input stream is done

### PR DESCRIPTION
**Pull request by chaosbox**
This PR fixes
 - the OOM caused by unnecessary creation of timer objects even after the request/response is completed. 
Find pprof attached that shows before and after the fix.

Before:
image

After:
image

### CODE REVIEW

**Comment by xthexder** [comment]
Thanks for the fix! < 2MB is definitely a much more reasonable amount of memory usage.

I'll take a closer look later tonight and get this merged.

**Comment by fantayeneh** [comment]
Ola @xthexder, can we merge this?

**Comment by chaosbox** [comment]
@jpittis @xthexder 
hi guys,
any feedback on this PR? 
thanks

**Comment by pgeorgiev333** [comment]
Can you guys merge this and create a new release? This is pretty important as we cannot simulate network failures for longer periods of time.
Thanks in advance

File: toxics/timeout.go [hunk 1]
```diff
@@ -19,7 +19,10 @@ func (t *TimeoutToxic) Pipe(stub *ToxicStub) {
 				return
 			case <-stub.Interrupt:
 				return
-			case <-stub.Input:
+			case c := <-stub.Input:
+				if c == nil {
+					return
```
Comment (jpittis) [inline toxics/timeout.go]: I'm pretty sure this needs a `stub.Close()`.

**Comment by jpittis** [comment]
So I'm pretty sure this is good to merge now.

Any chance you can sanity check @xthexder?

Unfortunately, our company recently rolled out this contributor license agreement stuff so I can't merge this until the original contributor has clicked through the agreement. @chaosbox 

I'm so sorry this has taken so long to ship, I'll release this as soon as we merge.

**Comment by jpittis** [comment]
I think you're going to have to change your commit to be authored by the same public email.

**Comment by xthexder** [comment]
This looks good to me.
I haven't really had much time to look over Toxiproxy recently, I may try to find some time over the holidays to do a pass and fix any outstanding issues.

**Pull request merged**